crates/swc_ecma_transforms_compat/src/es2015/arrow.rs

```diff
@@ -83,6 +83,7 @@ impl VisitMut for Arrow {
                 ..
             }) => {
                 params.visit_mut_with(self);
+                params.visit_mut_with(&mut FunctionWrapper::new(&mut self.state));
```

Comment:
Isn't this exponential?

Reply:
Nevermind. `n` will be super small.

Reply:
Could you explain why this is exponential?

Reply:
I was confused, so I made a benchmark that only profiles nested visitor.
You can see the content of benchmark at https://github.com/swc-project/swc/pull/2820/files and


```
    Finished bench [optimized + debuginfo] target(s) in 1m 20s
     Running unittests (/Users/kdy1/projects/s/issues/target/release/deps/time_complexity-81379e7c22c61e80)

running 6 tests
test time_10 ... bench:         238 ns/iter (+/- 33)
test time_15 ... bench:         480 ns/iter (+/- 27)
test time_20 ... bench:         874 ns/iter (+/- 29)
test time_40 ... bench:       5,086 ns/iter (+/- 212)
test time_5  ... bench:          71 ns/iter (+/- 3)
test time_60 ... bench:      13,085 ns/iter (+/- 480)

test result: ok. 0 passed; 0 failed; 0 ignored; 6 measured; 0 filtered out; finished in 2.80s
```


This is what I get.

Reply:
I see, maybe I could figure out a way to avoid this

Reply:
Wait, wouldn't swc translate the innermost arrow expression first, and every time a `FunctionWrapper` is started, it will be stopped to visit inner arrow by `FnExpr`?

Reply:
See the new test in the new commit, only 6 expressions is printed, I guess it's fine

Reply:
The number it printed does not matter.
At the benchmark above I posted, `visit count` is also tested.
It's exponential because it's deeply nested, not because a node is visited exponential times.


> Wait, wouldn't swc translate the innermost arrow expression first, and every time a FunctionWrapper is started, it will be stopped to visit inner arrow by FnExpr?

About this, I think it will be fine for this line, but not for `body` below
https://github.com/swc-project/swc/blob/9221b0744bc921957bd9a491f1499f55fa5c9443/crates/swc_ecma_transforms_compat/src/es2015/arrow.rs#L100

Reply:
Ok, let's move this lengthy discussion to github discussion